### PR TITLE
Added node revisions support

### PR DESCRIPTION
**Pull request by jocelyn**

### CODE REVIEW

**Comment by jocelyn** [comment]
Do not merge it for now.
This is mainly for code review.
And if ok, I will merge it.

**Comment by jocelyn** [comment]
I will merge it myself on Monday.
But for now, I would appreciate feedback

the idea is to keep 
nodes table as it is  (only added a constraint for nid+revision to be unique)

and then, revisions goes to node_revisions table, with only relevant fields.

for extensions like blog, page .. the table has all revisions in same blog_node_posts table, and related.
The revision increments per node .. (i.e node id, and rev from 1 to ... ) as discussed before (i.e solution 2)

Ideally we should review the nodes table, to have less fields, and always have an entry in node_revisions . And we could move the "content" there .. this way, when we want to load a list of nodes quickly we would not load eventual big 'content' ... 

but for now, this would break existing sites, so let's postponed such improvement to later .. once we have a smart code+db update solution.

File: modules/blog/cms_node_storage_sql_blog_extension.e [hunk 1]
```diff
@@ -68,37 +72,60 @@ feature -- Persistence
...
+						-- no page data, means everything is empty.
+						-- FOR NOW: always record row
+--						sql_change (sql_insert_node_data, l_parameters)
+					end
 				end
 			end
 		end
 
 	load (a_node: CMS_BLOG)
 		local
```
Comment (jvelilla) [inline modules/blog/cms_node_storage_sql_blog_extension.e]: Missing description.

File: modules/blog/cms_node_storage_sql_blog_extension.e [hunk 2]
```diff
@@ -68,37 +72,60 @@ feature -- Persistence
...
+			l_tags: READABLE_STRING_32
+		do
+			if attached node_data (a_node) as d then
+				l_tags := d.tags
+				a_node.set_tags_from_string (l_tags)
+			end
+		end
+
+	node_data (a_node: CMS_NODE): detachable TUPLE [revision: INTEGER_64; tags: READABLE_STRING_32]
+		local
```
Comment (jvelilla) [inline modules/blog/cms_node_storage_sql_blog_extension.e]: Description missing.

File: modules/node/cms_node_api.e [hunk 3]
```diff
@@ -220,6 +221,12 @@ feature -- Access: Node
 			Result := node_storage.nodes
 		end
 
+	node_revisions (a_node: CMS_NODE): LIST [CMS_NODE]
```
Comment (jvelilla) [inline modules/node/cms_node_api.e]: Description missing.  Possibly list or node revisions for a given node `a_node'.

File: modules/blog/cms_node_storage_sql_blog_extension.e (hunk 1 above)
Comment (jocelyn) [inline modules/blog/cms_node_storage_sql_blog_extension.e]: done

File: modules/blog/cms_node_storage_sql_blog_extension.e (hunk 2 above)
Comment (jocelyn) [inline modules/blog/cms_node_storage_sql_blog_extension.e]: done

File: modules/node/cms_node_api.e (hunk 3 above)
Comment (jocelyn) [inline modules/node/cms_node_api.e]: done

**Pull request merged**